// ngx-stars/bundles/ngx-stars.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/common'),exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ngx-stars', ['@angular/core','@angular/common','exports', '@angular/core', '@angular/common'], factory) :
    (factory(global.ng.core,global.ng.common,(global['ngx-stars'] = {}),global.ng.core,global.ng.common));
}(this, (function (ɵngcc0,ɵngcc1,exports,core,common) { 
function NgxStarsComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    var _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 3);
    ɵngcc0.ɵɵlistener("click", function NgxStarsComponent_div_2_Template_div_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r3); var star_r1 = ctx.$implicit; var ctx_r2 = ɵngcc0.ɵɵnextContext(); return ctx_r2.readonly ? ctx_r2.noop() : ctx_r2.onStarClick($event, star_r1); })("mousemove", function NgxStarsComponent_div_2_Template_div_mousemove_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r3); var star_r1 = ctx.$implicit; var ctx_r4 = ɵngcc0.ɵɵnextContext(); return ctx_r4.readonly ? ctx_r4.noop() : ctx_r4.onStarHover($event, star_r1); });
    ɵngcc0.ɵɵelement(1, "span", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var star_r1 = ctx.$implicit;
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ctx_r0.starPadding());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", star_r1.classname)("ngStyle", ctx_r0.starColorAndSize());
} }
'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var NgxStarsComponent = /** @class */ (function () {
        function NgxStarsComponent() {
            var _this = this;
            this.maxStars = 5;
            this.initialStars = 0;
            this.animationSpeed = 100;
            this.wholeStars = false;
            this.ratingOutput = new core.EventEmitter();
            this.customClassIdentifier = Math.random().toString(36).substring(2);
            this.safeSize = function () { return (Number.isInteger(_this.size) && _this.size > 0 && _this.size < 6) ? _this.size : 1; };
        }
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.setupStarImages();
                this.editableStars = Array.from(new Array(this.maxStars)).map(function (elem, index) { return new EditableStar(index); });
                this.setRating(this.initialStars);
                if (this.animation) {
                    this.animationInterval = setInterval(this.starAnimation.bind(this), this.animationSpeed);
                }
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.ngOnDestroy = /**
         * @return {?}
         */
            function () {
                // remove the three custom classes we created if custom image urls were provided
                if (this.customCssClasses) {
                    this.customCssClasses.forEach(function (style) {
                        if (style && style.parentNode) {
                            style.parentNode.removeChild(style);
                        }
                    });
                }
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.setupStarImages = /**
         * @return {?}
         */
            function () {
                var _this = this;
                if (this.customStarIcons) {
                    this.customCssClasses = [];
                    Object.keys(this.customStarIcons).map(function (key) { return (key); }).forEach(function (starType) {
                        /** @type {?} */
                        var classname = _this.getStarClass(starType);
                        _this.createCssClass(classname, starType);
                    });
                }
            };
        /**
         * @param {?} classname
         * @param {?} starType
         * @return {?}
         */
        NgxStarsComponent.prototype.createCssClass = /**
         * @param {?} classname
         * @param {?} starType
         * @return {?}
         */
            function (classname, starType) {
                /** @type {?} */
                var clazz = document.createElement('style');
                clazz.type = 'text/css';
                clazz.innerHTML = "." + classname + " {\n      -webkit-mask-image: url(" + this.customStarIcons[starType] + ");\n      mask-image: url(" + this.customStarIcons[starType] + ");\n    }";
                document.getElementsByTagName('head')[0].appendChild(clazz);
                this.customCssClasses.push(clazz);
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.starPadding = /**
         * @return {?}
         */
            function () {
                return { 'margin-right': this.customPadding || "0." + this.safeSize() + "rem" };
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.starColorAndSize = /**
         * @return {?}
         */
            function () {
                return Object.assign({}, this.starColor(), this.starSize());
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.starColor = /**
         * @return {?}
         */
            function () {
                return { 'background-color': this.color || 'crimson' };
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.starSize = /**
         * @return {?}
         */
            function () {
                return {
                    height: 15 * this.safeSize() + "px",
                    width: 16 * this.safeSize() + "px",
                };
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.starAnimation = /**
         * @return {?}
         */
            function () {
                this.animationRunning = true;
                if (this.rating < this.maxStars) {
                    this.setRating(this.rating += 0.5);
                }
                else {
                    this.setRating(0);
                }
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.cancelStarAnimation = /**
         * @return {?}
         */
            function () {
                if (this.animationRunning) {
                    clearInterval(this.animationInterval);
                    this.rating = 0;
                    this.animationRunning = false;
                }
            };
        /**
         * @param {?} rating
         * @return {?}
         */
        NgxStarsComponent.prototype.setRating = /**
         * @param {?} rating
         * @return {?}
         */
            function (rating) {
                this.rating = Math.round(rating * 2) / 2;
                this.onStarsUnhover();
            };
        /**
         * @param {?} event
         * @param {?} clickedStar
         * @return {?}
         */
        NgxStarsComponent.prototype.onStarHover = /**
         * @param {?} event
         * @param {?} clickedStar
         * @return {?}
         */
            function (event, clickedStar) {
                var _this = this;
                this.cancelStarAnimation();
                /** @type {?} */
                var clickedInFirstHalf = this.clickedInFirstHalf(event);
                // fill in either a half or whole star depending on where user clicked
                clickedStar.classname = (!this.wholeStars && clickedInFirstHalf) ? this.getStarClass('half') : this.getStarClass('full');
                // fill in all stars in previous positions and clear all in later ones
                this.editableStars.forEach(function (star) {
                    if (star.position > clickedStar.position) {
                        star.classname = _this.getStarClass('empty');
                    }
                    else if (star.position < clickedStar.position) {
                        star.classname = _this.getStarClass('full');
                    }
                });
            };
        /**
         * @param {?} event
         * @param {?} clickedStar
         * @return {?}
         */
        NgxStarsComponent.prototype.onStarClick = /**
         * @param {?} event
         * @param {?} clickedStar
         * @return {?}
         */
            function (event, clickedStar) {
                this.cancelStarAnimation();
                /** @type {?} */
                var clickedInFirstHalf = this.clickedInFirstHalf(event);
                this.rating = clickedStar.position + ((!this.wholeStars && clickedInFirstHalf) ? 0.5 : 1);
                this.ratingOutput.emit(this.rating);
            };
        // hidden star to left of first star lets user click there to set to 0
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.onZeroStarClick = /**
         * @return {?}
         */
            function () {
                this.setRating(0);
                this.ratingOutput.emit(this.rating);
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.onZeroStarHover = /**
         * @return {?}
         */
            function () {
                var _this = this;
                // clear all stars
                this.editableStars.forEach(function (star) { return star.classname = _this.getStarClass('empty'); });
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.onStarsUnhover = /**
         * @return {?}
         */
            function () {
                var _this = this;
                // when user stops hovering we want to make stars reflect the last rating applied by clicking
                this.editableStars.forEach(function (star) {
                    /** @type {?} */
                    var starNumber = star.position + 1;
                    if (_this.rating >= starNumber) {
                        star.classname = _this.getStarClass('full');
                    }
                    else if (_this.rating > starNumber - 1 && _this.rating < starNumber) {
                        star.classname = _this.getStarClass('half');
                    }
                    else {
                        star.classname = _this.getStarClass('empty');
                    }
                });
            };
        /**
         * @param {?} event
         * @return {?}
         */
        NgxStarsComponent.prototype.clickedInFirstHalf = /**
         * @param {?} event
         * @return {?}
         */
            function (event) {
                /** @type {?} */
                var starIcon = /** @type {?} */ (event.target);
                return event.pageX < starIcon.getBoundingClientRect().left + starIcon.offsetWidth / 2;
            };
        /**
         * @return {?}
         */
        NgxStarsComponent.prototype.noop = /**
         * @return {?}
         */
            function () { };
        /**
         * @param {?} starType
         * @return {?}
         */
        NgxStarsComponent.prototype.getStarClass = /**
         * @param {?} starType
         * @return {?}
         */
            function (starType) {
                if (this.customCssClasses) {
                    return "ngx-stars-star-" + starType + "-" + this.customClassIdentifier;
                }
                return "star-" + starType;
            };
        NgxStarsComponent.propDecorators = {
            maxStars: [{ type: core.Input }],
            initialStars: [{ type: core.Input }],
            readonly: [{ type: core.Input }],
            size: [{ type: core.Input }],
            color: [{ type: core.Input }],
            animation: [{ type: core.Input }],
            animationSpeed: [{ type: core.Input }],
            customPadding: [{ type: core.Input }],
            wholeStars: [{ type: core.Input }],
            customStarIcons: [{ type: core.Input }],
            ratingOutput: [{ type: core.Output }]
        };
NgxStarsComponent.ɵfac = function NgxStarsComponent_Factory(t) { return new (t || NgxStarsComponent)(); };
NgxStarsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NgxStarsComponent, selectors: [["ngx-stars"]], inputs: { maxStars: "maxStars", initialStars: "initialStars", animationSpeed: "animationSpeed", wholeStars: "wholeStars", readonly: "readonly", size: "size", color: "color", animation: "animation", customPadding: "customPadding", customStarIcons: "customStarIcons" }, outputs: { ratingOutput: "ratingOutput" }, decls: 3, vars: 2, consts: [[1, "stars-line", 3, "mouseleave"], ["aria-hidden", "true", 1, "star", "zero-star", 3, "ngStyle", "click", "mousemove"], [3, "ngStyle", "click", "mousemove", 4, "ngFor", "ngForOf"], [3, "ngStyle", "click", "mousemove"], ["aria-hidden", "true", 1, "star", 3, "ngClass", "ngStyle"]], template: function NgxStarsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵlistener("mouseleave", function NgxStarsComponent_Template_div_mouseleave_0_listener() { return ctx.readonly ? ctx.noop() : ctx.onStarsUnhover(); });
        ɵngcc0.ɵɵelementStart(1, "span", 1);
        ɵngcc0.ɵɵlistener("click", function NgxStarsComponent_Template_span_click_1_listener() { return ctx.onZeroStarClick(); })("mousemove", function NgxStarsComponent_Template_span_mousemove_1_listener() { return ctx.readonly ? ctx.noop() : ctx.onZeroStarHover(); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(2, NgxStarsComponent_div_2_Template, 2, 3, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngStyle", ctx.starSize());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.editableStars);
    } }, directives: [ɵngcc1.NgStyle, ɵngcc1.NgForOf, ɵngcc1.NgClass], styles: [".stars-line[_ngcontent-%COMP%]{display:flex;align-items:center;position:relative}.stars-line[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{z-index:999}.zero-star[_ngcontent-%COMP%]{color:transparent;position:absolute;left:-16px}.star[_ngcontent-%COMP%]{display:inline-block;-webkit-mask-repeat:no-repeat;mask-repeat:no-repeat}.star-empty[_ngcontent-%COMP%]{-webkit-mask-image:url(\"data:image/svg+xml,%3Csvg aria-hidden='true' focusable='false' data-prefix='far' data-icon='star' class='svg-inline--fa fa-star fa-w-18' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 576 512'%3E%3Cpath fill='currentColor' d='M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z'%3E%3C/path%3E%3C/svg%3E%0A\");mask-image:url(\"data:image/svg+xml,%3Csvg aria-hidden='true' focusable='false' data-prefix='far' data-icon='star' class='svg-inline--fa fa-star fa-w-18' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 576 512'%3E%3Cpath fill='currentColor' d='M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z'%3E%3C/path%3E%3C/svg%3E%0A\")}.star-half[_ngcontent-%COMP%]{-webkit-mask-image:url(\"data:image/svg+xml,%3C!-- had to hack this one's viewbox otherwise it didn't line up with the other two --%3E%3C!-- changed viewbox from '0 0 536 512' to '-20 0 576 512' --%3E%3Csvg aria-hidden='true' focusable='false' data-prefix='fas' data-icon='star-half-alt' class='svg-inline--fa fa-star-half-alt fa-w-17' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='-20 0 576 512'%3E%3Cpath fill='currentColor' d='M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z'%3E%3C/path%3E%3C/svg%3E\");mask-image:url(\"data:image/svg+xml,%3C!-- had to hack this one's viewbox otherwise it didn't line up with the other two --%3E%3C!-- changed viewbox from '0 0 536 512' to '-20 0 576 512' --%3E%3Csvg aria-hidden='true' focusable='false' data-prefix='fas' data-icon='star-half-alt' class='svg-inline--fa fa-star-half-alt fa-w-17' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='-20 0 576 512'%3E%3Cpath fill='currentColor' d='M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z'%3E%3C/path%3E%3C/svg%3E\")}.star-full[_ngcontent-%COMP%]{-webkit-mask-image:url(\"data:image/svg+xml,%3Csvg aria-hidden='true' focusable='false' data-prefix='fas' data-icon='star' class='svg-inline--fa fa-star fa-w-18' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 576 512'%3E%3Cpath fill='currentColor' d='M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z'%3E%3C/path%3E%3C/svg%3E\");mask-image:url(\"data:image/svg+xml,%3Csvg aria-hidden='true' focusable='false' data-prefix='fas' data-icon='star' class='svg-inline--fa fa-star fa-w-18' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 576 512'%3E%3Cpath fill='currentColor' d='M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z'%3E%3C/path%3E%3C/svg%3E\")}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxStarsComponent, [{
        type: core.Component,
        args: [{
                selector: 'ngx-stars',
                template: "<div class=\"stars-line\" (mouseleave)=\"readonly ? noop() : onStarsUnhover()\">\n  <span class=\"star zero-star\" [ngStyle]=\"starSize()\" aria-hidden=\"true\" (click)=\"onZeroStarClick()\" (mousemove)=\"readonly ? noop() : onZeroStarHover()\"></span>\n  <div *ngFor=\"let star of editableStars;\" [ngStyle]=\"starPadding()\" (click)=\"readonly ? noop() : onStarClick($event, star)\" (mousemove)=\"readonly ? noop() : onStarHover($event, star)\">\n    <span class=\"star\" [ngClass]=\"star.classname\" [ngStyle]=\"starColorAndSize()\" aria-hidden=\"true\"></span>\n  </div>\n</div>\n",
                styles: [".stars-line{display:flex;align-items:center;position:relative}.stars-line>div{z-index:999}.zero-star{color:transparent;position:absolute;left:-16px}.star{display:inline-block;-webkit-mask-repeat:no-repeat;mask-repeat:no-repeat}.star-empty{-webkit-mask-image:url(\"data:image/svg+xml,%3Csvg aria-hidden='true' focusable='false' data-prefix='far' data-icon='star' class='svg-inline--fa fa-star fa-w-18' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 576 512'%3E%3Cpath fill='currentColor' d='M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z'%3E%3C/path%3E%3C/svg%3E%0A\");mask-image:url(\"data:image/svg+xml,%3Csvg aria-hidden='true' focusable='false' data-prefix='far' data-icon='star' class='svg-inline--fa fa-star fa-w-18' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 576 512'%3E%3Cpath fill='currentColor' d='M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z'%3E%3C/path%3E%3C/svg%3E%0A\")}.star-half{-webkit-mask-image:url(\"data:image/svg+xml,%3C!-- had to hack this one's viewbox otherwise it didn't line up with the other two --%3E%3C!-- changed viewbox from '0 0 536 512' to '-20 0 576 512' --%3E%3Csvg aria-hidden='true' focusable='false' data-prefix='fas' data-icon='star-half-alt' class='svg-inline--fa fa-star-half-alt fa-w-17' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='-20 0 576 512'%3E%3Cpath fill='currentColor' d='M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z'%3E%3C/path%3E%3C/svg%3E\");mask-image:url(\"data:image/svg+xml,%3C!-- had to hack this one's viewbox otherwise it didn't line up with the other two --%3E%3C!-- changed viewbox from '0 0 536 512' to '-20 0 576 512' --%3E%3Csvg aria-hidden='true' focusable='false' data-prefix='fas' data-icon='star-half-alt' class='svg-inline--fa fa-star-half-alt fa-w-17' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='-20 0 576 512'%3E%3Cpath fill='currentColor' d='M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z'%3E%3C/path%3E%3C/svg%3E\")}.star-full{-webkit-mask-image:url(\"data:image/svg+xml,%3Csvg aria-hidden='true' focusable='false' data-prefix='fas' data-icon='star' class='svg-inline--fa fa-star fa-w-18' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 576 512'%3E%3Cpath fill='currentColor' d='M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z'%3E%3C/path%3E%3C/svg%3E\");mask-image:url(\"data:image/svg+xml,%3Csvg aria-hidden='true' focusable='false' data-prefix='fas' data-icon='star' class='svg-inline--fa fa-star fa-w-18' role='img' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 576 512'%3E%3Cpath fill='currentColor' d='M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z'%3E%3C/path%3E%3C/svg%3E\")}"]
            }]
    }], function () { return []; }, { maxStars: [{
            type: core.Input
        }], initialStars: [{
            type: core.Input
        }], animationSpeed: [{
            type: core.Input
        }], wholeStars: [{
            type: core.Input
        }], ratingOutput: [{
            type: core.Output
        }], readonly: [{
            type: core.Input
        }], size: [{
            type: core.Input
        }], color: [{
            type: core.Input
        }], animation: [{
            type: core.Input
        }], customPadding: [{
            type: core.Input
        }], customStarIcons: [{
            type: core.Input
        }] }); })();
        return NgxStarsComponent;
    }());
    var EditableStar = /** @class */ (function () {
        function EditableStar(position) {
            this.position = position;
        }
        return EditableStar;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var NgxStarsModule = /** @class */ (function () {
        function NgxStarsModule() {
        }
NgxStarsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxStarsModule });
NgxStarsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxStarsModule_Factory(t) { return new (t || NgxStarsModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxStarsModule, { declarations: [NgxStarsComponent], imports: [ɵngcc1.CommonModule], exports: [NgxStarsComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxStarsModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    NgxStarsComponent
                ],
                exports: [
                    NgxStarsComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return NgxStarsModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    exports.NgxStarsComponent = NgxStarsComponent;
    exports.EditableStar = EditableStar;
    exports.NgxStarsModule = NgxStarsModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
