// ngx-stars/esm2015/lib/ngx-stars.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { NgxStarsComponent } from './ngx-stars.component';
import { CommonModule } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
export class NgxStarsModule {
}
NgxStarsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxStarsModule });
NgxStarsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxStarsModule_Factory(t) { return new (t || NgxStarsModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxStarsModule, { declarations: function () { return [NgxStarsComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxStarsComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxStarsModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    NgxStarsComponent
                ],
                exports: [
                    NgxStarsComponent
                ]
            }]
    }], null, null); })();
