// ngx-mat-datetime-picker/esm5/lib/utils/date-utils.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/utils/date-utils.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
export var LIMIT_TIMES = {
    minHour: 0,
    maxHour: 24,
    minMinute: 0,
    maxMinute: 60,
    minSecond: 0,
    maxSecond: 60,
    meridian: 12
};
/** @type {?} */
export var MERIDIANS = {
    AM: 'AM',
    PM: 'PM'
};
/** @type {?} */
export var DEFAULT_STEP = 1;
/** @type {?} */
export var DEFAULT_HOUR_PLACEHOLDER = '';
/** @type {?} */
export var DEFAULT_MINUTE_PLACEHOLDER = '';
/** @type {?} */
export var DEFAULT_SECOND_PLACEHOLDER = '';
/** @type {?} */
export var NUMERIC_REGEX = /[^0-9]/g;
/** @type {?} */
export var PATTERN_INPUT_HOUR = /^(2[0-3]|[0-1][0-9]|[0-9])$/;
/** @type {?} */
export var PATTERN_INPUT_MINUTE = /^([0-5][0-9]|[0-9])$/;
/** @type {?} */
export var PATTERN_INPUT_SECOND = /^([0-5][0-9]|[0-9])$/;
/**
 * @param {?} val
 * @return {?}
 */
export function formatTwoDigitTimeValue(val) {
    /** @type {?} */
    var txt = val.toString();
    return txt.length > 1 ? txt : "0" + txt;
}
/**
 * @param {?} provider
 * @return {?}
 */
export function createMissingDateImplError(provider) {
    return Error("MatDatepicker: No provider found for " + provider + ". You must import one of the following " +
        "modules at your application root: NgxMatNativeDateModule, NgxMatMomentModule, or provide a " +
        "custom implementation.");
}
/**
 * Formats a range of years.
 * @param {?} start
 * @param {?} end
 * @return {?}
 */
export function formatYearRange(start, end) {
    return start + " \u2013 " + end;
}
